### ISSUE TITLE
Inconsistent output compared with other JS engines: syntax error on large integer literal in regexp

### ISSUE THREAD

**Issue by sunlili** [Microsoft/ChakraCore#5361]
Hello,
The following code behaves incorrectly (inconsistent with the standard and other engines).
ch version 1.8.5.0
```
(("").match(/(A{9999999999}B|C*)*D/));  
print("BT_FLAG");  
```
Output is:
SyntaxError: Syntax error in regular expression
	at code (55577775.js:1:26)

However, other engines output:
BT_FLAG

BT group
2018.6.25

**Comment by MSLaguana** [comment]
This looks like a duplicate of https://github.com/Microsoft/ChakraCore/issues/5182

**Comment by dilijev** [comment]
Confirmed the test case behaves as expected in release/1.10 as a result of the fix to #5182 (PR #5311)
